airflow/models/param.py

```diff
@@ -57,10 +59,28 @@ def __init__(self, default: Any = __NO_VALUE_SENTINEL, description: Optional[str
 
         # If we have a value, validate it once. May raise ValueError.
         if self.has_value:
-            try:
-                jsonschema.validate(self.value, self.schema, format_checker=FormatChecker())
-            except ValidationError as err:
-                raise ValueError(err)
+            self._validate(self.value, self.schema)
+
+    @staticmethod
+    def _validate(value, schema):
+        """
+        1. Check that value is json-serializable; if not, warn.  In future release we will require
+        the value to be json-serializable.
+        2. Validate ``value`` against ``schema``
+        """
+        try:
+            json.dumps(value)
+        except Exception:
+            warnings.warn(
+                "The use of non-json-serializable params is deprecated and will be removed in "
+                " a future release",
+                DeprecationWarning,
+                stacklevel=2,
```

Comment:
Is this deep enough?

Reply:
Do you mean the stackllevel param? I just used what we use elsewhere...

Reply:
nudge?

Reply:
Yeah, stacklevel. We want it to point to someplace meaningful for the user, otherwise we shouldn't skip any.

(sorry, not sure how I missed this last week)